packages/components/tests/dummy/app/templates/components/code-block.hbs

```diff
@@ -0,0 +1,510 @@
+  console.log(`ACTUAL: ${actualStr}  EXPECTED: ${expectedStr}`);
+  if (actualStr === expectedStr) {
+    console.log('passed');
+  } else {
+    console.log(`FAILED [${testName}] Expected ${JSON.stringify(expected)}, but got ${JSON.stringify(actual)}`);
+  }
+}"
+        as |CB|
+      >
+        <CB.Title>Title</CB.Title>
+        <CB.Description>Description</CB.Description>
+      </Hds::CodeBlock>
+      {{! template-lint-enable no-whitespace-for-layout }}
+    </SG.Item>
+  </Shw::Flex>
+
+  <Shw::Divider @level="2" />
+
+  <Shw::Text::H3>Editable content</Shw::Text::H3>
+
+  <Shw::Flex @columns={{2}} {{style gap="2rem"}} as |SG|>
+    <SG.Item {{style width="calc(50% - 1rem)"}} @label="contentEditable=true">
+      <Hds::CodeBlock
+        @language="javascript"
+        @contentEditable={{true}}
+        @onInput={{this.noop}}
+        @value="let codeLang='JavaScript';
+console.log(`I am ${codeLang} code`);"
+      />
+    </SG.Item>
+
+    <SG.Item
+      {{style width="calc(50% - 1rem)"}}
+      @label="contentEditable=true, hasCopyButton=true, title and description"
+    >
+      <Hds::CodeBlock
+        id="clipboardTarget3"
+        @contentEditable={{true}}
+        @hasCopyButton={{true}}
+        @language="go"
+        @value="package main
+import 'fmt'
+func main() {
+  fmt.Println('hello world')
+}"
+        as |CB|
+      >
+        <CB.Title>Title</CB.Title>
+        <CB.Description>Description</CB.Description>
+      </Hds::CodeBlock>
+    </SG.Item>
+  </Shw::Flex>
+
+  <Shw::Divider @level="2" />
+
+  <Shw::Text::H3>Highlight lines</Shw::Text::H3>
+
+  <Shw::Flex @columns={{2}} {{style gap="2rem"}} as |SG|>
+    <SG.Item {{style width="calc(50% - 1rem)"}} @label="Highlight lines 2 & 4">
+      <Hds::CodeBlock
+        @language="javascript"
+        @highlightLines={{"2, 4"}}
+        @value="import Application from '@ember/application';
+import Resolver from 'ember-resolver';
+import loadInitializers from 'ember-load-initializers';
+import config from 'dummy/config/environment';
+
+export default class App extends Application {
+  modulePrefix = config.modulePrefix;
+  podModulePrefix = config.podModulePrefix;
+  Resolver = Resolver;
+}
+
+loadInitializers(App, config.modulePrefix);"
+      />
+    </SG.Item>
+
+    <SG.Item {{style width="calc(50% - 1rem)"}} @label="Highlight lines 6-10">
+      <Hds::CodeBlock
+        @language="javascript"
+        @highlightLines={{"6-10"}}
+        @value="import Application from '@ember/application';
+import Resolver from 'ember-resolver';
+import loadInitializers from 'ember-load-initializers';
+import config from 'dummy/config/environment';
+
+export default class App extends Application {
+  modulePrefix = config.modulePrefix;
+  podModulePrefix = config.podModulePrefix;
+  Resolver = Resolver;
+}
+
+loadInitializers(App, config.modulePrefix);"
+      />
+    </SG.Item>
+  </Shw::Flex>
+
+  <Shw::Divider @level="2" />
+  <Shw::Text::H3>Language</Shw::Text::H3>
```

Comment:
yes, I made sure it fails gracefully as [part of the setup](https://github.com/hashicorp/design-system/pull/1687/files#diff-54fbcfe4d8d03db33af48acaa4408fe0b24c37526daf2556b17bef1d5a0cdfa9R134). Will add an example to prevent regression.